controller/src/main/java/org/jboss/as/controller/AbstractAddStepHandler.java

```diff
@@ -274,6 +275,11 @@ protected void recordCapabilitiesAndRequirements(OperationContext context, Model
                 context.registerCapability(capability, null);
             }
         }
+
+        //just here until we fix the registration across whole server
+        if (context.getResourceRegistration().getCapabilities().size() != capabilities.size()){
+            ControllerLogger.ROOT_LOGGER.warnf("Number of possible capabilities registered on resource and number of runtime capabilities don't match for path: %s", context.getCurrentAddress());
+        }
```

Comment:
This is ok, but please file a Blocker JIRA to get rid of this by CR1 so it doesn't fall into a crack.